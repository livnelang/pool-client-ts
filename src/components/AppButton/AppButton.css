.AppButton {
    display: flex;
    justify-content: center;
    align-items: center;
    color: #FFFFFFFF;
    padding: 19px 32px;
    border-radius: 32px;
    box-sizing: border-box;
    width: 100%;
    text-align: center;
}

.AppButton.primary {
    background: var(--cta);
    color: #FFFFFFFF;
}

.AppButton.grey {
    background: #F4F5F7;
    color: var(--main);
}

.AppButton.transparent {
    background: transparent;
    border: 1px solid #D0DBEA;
    color: var(--main);
}

.AppButton.regular {
    padding: 19px 32px;
    border-radius: 32px;
}

.AppButton.small {
    padding: 8px 16px;
    border-radius: 8px;
}