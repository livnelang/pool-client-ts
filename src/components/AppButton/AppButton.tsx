import "./AppButton.css";

export type AppButtonVariant = "primary" | "grey" | "transparent";

export interface AppButtonProps {
  text: string;
  onClick?: () => void;
  variant?: AppButtonVariant;
  children?: React.ReactNode;
  disabled?: boolean;
}

const AppButton = (props: AppButtonProps) => {
  return (
    <div
      style={props.disabled ? { pointerEvents: "none", opacity: "0.7" } : {}}
      className={`AppButton ${props.variant || "primary"}`}
      onClick={props.onClick}
    >
      <>
        {props.text}
        {props.children ?? null}
      </>
    </div>
  );
};

export default AppButton;
