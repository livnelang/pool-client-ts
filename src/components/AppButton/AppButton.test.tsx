import { describe, expect, it, test } from "vitest";
import renderer from 'react-test-renderer';
import AppButton from "./AppButton";

function toJson(component: renderer.ReactTestRenderer) {
    const result = component.toJSON()
    expect(result).toBeDefined()
    expect(result).not.toBeInstanceOf(Array)
    return result as renderer.ReactTestRendererJSON
  }
  
  test('Link changes the class when hovered', () => {
    const component = renderer.create(

      <AppButton text="Button Test"/>
    )
    let tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })