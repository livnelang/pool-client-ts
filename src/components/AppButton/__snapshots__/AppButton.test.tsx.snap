// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<div
  className="AppButton primary"
>
  Button Test
</div>
`;
